File: 997.find-the-town-judge.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

/*
 * @lc app=leetcode id=997 lang=java
 *
 * [997] Find the Town Judge
 */

// @lc code=start
class Solution {
    public int findJudge(int n, int[][] trust) {

       int[] trusting = new int[n+1];
       int[] trusted = new int[n+1];
      for (int[] relation : trust) {
          trusting[relation[0]]++;
          trusted[relation[1]]++;
      }

      for (int i = 1; i < trusted.length; i++) {
          if(trusted[i]==n-1 && trusting[i]==0){    
return i;
          }
      }
      return -1;
    }

}
// @lc code=end

File: 997.find-the-town-judge.py
#
# @lc app=leetcode id=997 lang=python3
#
# [997] Find the Town Judge
#

# @lc code=start
from typing import List


class Solution:
    
            
    
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        if n==1 and not trust:
            return 1
        
        count = [0 for _ in range(n + 1)]
        count2 = [0 for _ in range(n + 1)]
        for relation in trust:
            count[relation[1]] += 1
            count2[relation[0]] += 1
        if n - 1 in count:
            suspect = count.index(n - 1)
            return suspect if count2[suspect] == 0 else -1
        else:
            return -1

        
# @lc code=end

